### PR TITLE
Switch to html template

### PR DESCRIPTION
**Pull request by jeremylweiss**
full conversion of Shiny app to HTML-template approach

### CODE REVIEW

File: app/azmet-shiny-template.html [hunk 1]
```diff
@@ -0,0 +1,384 @@
...
+    <style>.shiny-notification {background-color: rgba(129,211,235,0.9) !important; border: none !important; box-shadow: 1px 2px rgba(0,0,0,0.1) !important; color: #1E5288 !important; font-style: italic !important; font-weight: bold !important; position: fixed !important; top: calc(50%) !important; left: calc(50% - 100px) !important;}</style>
+    <!-- id -->
+    <style>#downloadTSV {background-color: transparent; border-color: #001C48; color: #001C48; margin-bottom: 4px; margin-left: 4px; margin-top: 4px;}</style>
+    <style>#downloadTSV:active {background-color: #001C48; color: #fff;}</style>
+    <style>#downloadTSV:hover {background-color: #001C48; color: #fff;}</style>
+    <style>#previewData {background-color: transparent; border-color: #001C48; color: #001C48;}</style>
+    <style>#previewData:active {background-color: #001C48; color: #fff;}</style>
+    <style>#previewData:hover {background-color: #001C48; color: #fff;}</style>
+    <style>#sidebarPanel {background-color: #E2E9EB !important; margin-bottom: 72px}</style>
```
Comment (Aariq) [inline app/azmet-shiny-template.html:101]: I _think_ it's better practice to put this in a .css file, but I think that's more just for ease of editing.

File: app/azmet-shiny-template.html [hunk 2]
```diff
@@ -0,0 +1,384 @@
...
+                      </div>
+                    </section>
+                  </main>
+                </div>
+              </div>
+            </div>
+            <div class="container">
+              <!-- div -->
+                <!-- Shiny app -->
+                {{sidebarLayout}}
```
Comment (Aariq) [inline app/azmet-shiny-template.html:302]:
```suggestion
                {{ sidebarLayout }}
```
Just a style suggestion, based on the Shiny template documentation

Comment (Aariq) [reply app/azmet-shiny-template.html:302]: This basically just sticks the entire Shiny app in in this place?  That's super nice!

**Review by Aariq — COMMENTED** [review]
I think this makes the app.R code a lot easier to read, and although the HTML is still difficult for me to interpret, it's clear where the shiny app generated HTML gets inserted (between header and footer).  End result looks very nice too!  I made just a few minor comments/suggestions

File: app/app.R [hunk 3]
```diff
@@ -195,41 +154,28 @@ server <- function(input, output, session) {
...
-      )
-    }
-  
+    validate(
+      need(
+        input$startDate <= input$endDate, 
+        "Please select a 'Start Date' that is earlier than or the same as the 'End Date'."
+      ),
+      errorClass = "datepicker"
+    )
```
[the remark this answers is not in the document]
Comment (jeremylweiss) [reply app/app.R:177]: I started to read about it here: https://shiny.posit.co/r/articles/improve/validation/

File: app/azmet-shiny-template.html [hunk 4]
```diff
@@ -0,0 +1,384 @@
+<!DOCTYPE html>
+<html lang="en" dir="ltr" prefix="content: http://purl.org/rss/1.0/modules/content/  dc: http://purl.org/dc/terms/  foaf: http://xmlns.com/foaf/0.1/  og: http://ogp.me/ns#  rdfs: http://www.w3.org/2000/01/rdf-schema#  schema: http://schema.org/  sioc: http://rdfs.org/sioc/ns#  sioct: http://rdfs.org/sioc/types#  skos: http://www.w3.org/2004/02/skos/core#  xsd: http://www.w3.org/2001/XMLSchema# " class="sticky-footer">
+  <head>
+    <meta charset="utf-8" />
+    <meta name="Generator" content="Drupal 9 (https://www.drupal.org)" />
+    <meta name="MobileOptimized" content="width" />
+    <meta name="HandheldFriendly" content="true" />
+    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no" />
+    <meta http-equiv="x-ua-compatible" content="ie=edge" />
+    <link rel="shortcut icon" href="https://staging.azmet.arizona.edu/profiles/custom/az_quickstart/themes/custom/az_barrio/favicon.ico" type="image/vnd.microsoft.icon" />
```
[the remark this answers is not in the document]
Comment (jeremylweiss) [reply app/azmet-shiny-template.html:10]: Right, we'll do that once the staging site goes "live"

File: app/azmet-shiny-template.html [hunk 5]
```diff
@@ -0,0 +1,384 @@
...
+    <!-- link rel="canonical" href="https://staging.azmet.arizona.edu/past-data" /-->
+    <!-- link rel="shortlink" href="https://staging.azmet.arizona.edu/node/6" /-->
+    <!-- link rel="revision" href="https://staging.azmet.arizona.edu/past-data" /-->
+    <script src="https://staging.azmet.arizona.edu/sites/default/files/google_tag/az_comprehensive/google_tag.script.js?s0j7xh" defer></script>
```
[the remark this answers is not in the document]
Comment (jeremylweiss) [reply app/azmet-shiny-template.html:14]: I'm looking into this with CCT web folks . . .

[the remark this answers is not in the document]
Comment (jeremylweiss) [reply app/azmet-shiny-template.html:14]: I heard that this is added to all UA Quickstart sites by default and it’s managed by the central Marketing and Brand Management team. That being so, my plan is to add a new google analytics `<script>` to each app

File: app/azmet-shiny-template.html (hunk 1 above)
Comment (jeremylweiss) [reply app/azmet-shiny-template.html:101]: Good idea. Useful reference for getting started with CSS: https://developer.mozilla.org/en-US/docs/Learn/Getting_started_with_the_web/CSS_basics

**Pull request merged**